lib/src/local_working_copy.rs

```diff
@@ -1218,6 +1212,44 @@ impl FileSnapshotter<'_> {
         Ok(())
     }
 
+    /// Emits file paths that don't exist in the `present_entries`.
+    fn emit_deleted_files(
+        &self,
+        dir: &RepoPath,
+        file_states: FileStates<'_>,
+        present_entries: &PresentDirEntries,
+    ) {
+        #[derive(Clone, Copy, Debug, Eq, PartialEq)]
+        enum Kind {
+            Dir,
+            File,
+        }
+
+        let file_state_chunks = file_states.iter().chunk_by(|(path, _state)| {
+            // Extract <name> from <dir>, <dir>/<name>, or <dir>/<name>/**.
+            // (file_states may contain <dir> file on file->dir transition.)
+            debug_assert!(path.starts_with(dir));
+            let len = dir.as_internal_file_string().len() + (!dir.is_root() as usize);
```

Comment:
question: Regarding `(!dir.is_root() as usize)`, is this safe to do on Windows? In those cases, could the root be e.g. `C:\` and have length >1?

Reply:
It's a (normalized) `RepoPath`, so there's no platform difference.

Reply:
nit (non-blocking): We could write `if dir.is_root() { "/".len() } else { 0 }` to make it clearer what's happening here, but not a big deal regardless.